fix: allow semicolons if at the end of sql

jameselsey opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->
Fixes

## Summary

### Changes

Minor change to the `SUSPICIOUS_PATTERNS` regex in the Postgres MCP server, from `r'(?i);'` -> ` r'(?i);(?=\s*\S)'`

This was previously blocking simple queries like `select count(*) from my_table;` due to the trailing semi colon

### User experience

Before this change, I wasn't able to execute simple select queries against my DB because the `check_sql_injection_risk` function was blocking the queries.
I raised an issue but decided it was simple enough I could fix : https://github.com/awslabs/mcp/issues/469

I have tested this locally by adjusting my mcp configuration like this:
```
stdio_mcp_client = MCPClient(lambda: stdio_client(
    StdioServerParameters(
        command="uvx",
        args=[
            "--from", "/Users/jameselsey/dev/projects/misc/mcp/src/postgres-mcp-server",
            "awslabs.postgres-mcp-server",
            "--resource_arn", "my_resource_arn",
            "--secret_arn", "my_secret_arn",
            "--database", "mydb",
            "--region", "us-west-2",
            "--readonly", "True"
        ],
        env= {
            "AWS_PROFILE": "myprofile",
            "AWS_REGION": "us-west-2",
            "FASTMCP_LOG_LEVEL": "ERROR"
        },
        cwd="/Users/jameselsey/dev/projects/misc/mcp/src/postgres-mcp-server",
    )
))
``` 

with logs:
```
DEBUG | strands.tools.mcp.mcp_client | [Thread: MainThread, Session: 440e9649-4651-44a4-a870-cb9d51c062e3] calling MCP tool 'run_query' synchronously with tool_use_id=tooluse_tr3OGGIWRNyam2QRBeqE-w
2025-05-30 22:21:28.025 | INFO     | awslabs.postgres_mcp_server.server:run_query:231 - run_query: readonly:True, SQL:SELECT COUNT(*) AS total_transactions FROM transaction;
2025-05-30 22:21:29.044 | SUCCESS  | awslabs.postgres_mcp_server.server:run_query:253 - run_query successfully executed query:SELECT COUNT(*) AS total_transactions FROM transaction;
```

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x ] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [x ] I have performed a self-review of this change
* [ x] Changes have been tested
* [ ] Changes are documented

Is this a breaking change? (Y/N)

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

kennthhz commented on src/postgres-mcp-server/awslabs/postgres_mcp_server/mutable_sql_detector.py:
```diff
@@ -56,7 +56,7 @@
     r'(?i)\btruncate\b',  # TRUNCATE
     r'(?i)\bgrant\b|\brevoke\b',  # GRANT or REVOKE
-    r'(?i);',  # stacked queries
+    r'(?i);(?=\s*\S)',  # stacked queries
```
can you also make the similar fix in the same file under mysql_mcp_server? also you may need to update the test case

kennthhz commented on src/postgres-mcp-server/awslabs/postgres_mcp_server/mutable_sql_detector.py:
```diff
@@ -56,7 +56,7 @@
     r'(?i)\btruncate\b',  # TRUNCATE
     r'(?i)\bgrant\b|\brevoke\b',  # GRANT or REVOKE
-    r'(?i);',  # stacked queries
+    r'(?i);(?=\s*\S)',  # stacked queries
```
can you also make the similar fix in the same file under mysql_mcp_server? also you may need to update the test case

jameselsey commented:
Hi @kennthhz , I added some extra tests to cover some variations with comments. 
I dropped the suspicious pattern for single & multi line because they would match comments anywhere, and my new pattern covered this better.

I applied the same changes to mysql, and ran the tests for both locally

```
Name                                                  Stmts   Miss Branch BrPart  Cover   Missing
-------------------------------------------------------------------------------------------------
awslabs/__init__.py                                       0      0      0      0   100%
awslabs/postgres_mcp_server/__init__.py                   1      0      0      0   100%
awslabs/postgres_mcp_server/mutable_sql_detector.py      15      0      4      0   100%
awslabs/postgres_mcp_server/server.py                   141     10     34      6    91%   68, 77, 100, 134, 161, 347, 359, 403-405
-------------------------------------------------------------------------------------------------
TOTAL                                                   157     10     38      6    92%
======================================================================================================= 10 passed in 0.35s ========================================================================================================


Name                                               Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------------------------------------
awslabs/__init__.py                                    0      0      0      0   100%
awslabs/mysql_mcp_server/__init__.py                   1      0      0      0   100%
awslabs/mysql_mcp_server/mutable_sql_detector.py      26      0     12      0   100%
awslabs/mysql_mcp_server/server.py                   122      8     28      4    92%   67, 76, 99, 133, 160, 344-346
----------------------------------------------------------------------------------------------
TOTAL                                                149      8     40      4    94%
======================================================================================================= 10 passed in 0.95s ========================================================================================================


```

dotmanila commented:
If the MCP get `SELECT 1; SELECT 2;` I assume this will fail as a whole, can we splice the first `SELECT 1` and execute it, and send a warning that the second `SELECT` is ignored?
